protostar/starknet/forkable_starknet.py

```diff
@@ -51,14 +54,32 @@ async def deploy(
         constructor_calldata: Optional[List[int]] = None,
         disable_hint_validation: bool = False,
     ) -> StarknetContract:
-        starknet_contract = await super().deploy(
+        contract_class = get_contract_class(
             source=source,
             contract_class=contract_class,
-            contract_address_salt=contract_address_salt,
             cairo_path=cairo_path,
-            constructor_calldata=constructor_calldata,
             disable_hint_validation=disable_hint_validation,
         )
+        address, execution_info = await self.state.deploy(
+            contract_class=contract_class,
+            contract_address_salt=contract_address_salt,
+            constructor_calldata=[]
+            if constructor_calldata is None
+            else constructor_calldata,
+        )
+
+        deploy_call_info = StarknetCallInfo.from_internal(
+            call_info=as_non_optional(execution_info.call_info),
+            result=(),
+            main_call_events=[],
+        )
+
+        starknet_contract = StarknetContract(
+            state=self.state,
+            abi=get_abi(contract_class=contract_class),
+            contract_address=address,
+            deploy_call_info=deploy_call_info,
+        )
```

Comment:
Why is this code copied over from superclass?

Reply:
it's removed in [new approach](https://github.com/software-mansion/protostar/pull/1013/commits/3efa247b9c0b19218279da5b017f02f38aff4ebf)